extruct/jsonld.py

```diff
@@ -28,14 +28,23 @@ def extract_items(self, document, base_url=None):
             if items for item in items if item
         ]
 
+    def _may_be_get_json(self, script):
+        try:
+            return json.loads(script, strict=False)
+        except Exception:
+            return False
+
     def _extract_items(self, node):
         script = node.xpath('string()')
-        try:
-            # TODO: `strict=False` can be configurable if needed
-            data = json.loads(script, strict=False)
-        except ValueError:
-            # sometimes JSON-decoding errors are due to leading HTML or JavaScript comments
-            data = jstyleson.loads(HTML_OR_JS_COMMENTLINE.sub('', script), strict=False)
+        data = self._may_be_get_json(script)
+        # check if valid json.
+        if not data:
+            script = jstyleson.dispose( HTML_OR_JS_COMMENTLINE.sub('', script))
+            data = self._may_be_get_json(script)
+        # After processing check if json is still valid.
+        if not data:
+            return False
```

Comment:
Nitpick - we don't use return value, as this is a generator, so ``return`` would make more sense
```suggestion
            return
```

Reply:
fixed.